x-pack/plugins/osquery/server/handlers/action/get_action_type.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { curry, uniq } from 'lodash';
+import { ActionTypeExecutorResult } from '@kbn/actions-plugin/common';
+import { osqueryActionTypeBase } from '../../../common/actions/osquery_type';
+import { createActionHandler } from './create_action_handler';
+import { OsqueryAppContext } from '../../lib/osquery_app_context_services';
+
+interface IAlert {
+  agent?: {
+    id: string;
+  };
+}
+
+export const getOsqueryActionType = (osqueryContext: OsqueryAppContext) => ({
+  ...osqueryActionTypeBase,
+  executor: curry(executor)({
+    osqueryContext,
+  }),
+});
+
+// @ts-expect-error update types
+async function executor(payload, execOptions): Promise<ActionTypeExecutorResult<unknown>> {
+  const { query, ecs_mapping: ecsMapping, alerts, id } = execOptions.params;
```

Comment:
I see we're referencing `alerts` here.
Connectors are not meant to be limited to rules, so referencing the alerts domain in here could be a concern.
That said, this concern has come up recently in other places, so we've began discussing how to address this as a platform, and I don't see this as a blocker to this PR.

@mikecote - this chimes with the issue you opened yesterday about "feature specific connectors" (https://github.com/elastic/kibana/issues/135467). If this connector lands, it puts more urgency around making connector feature specific.

Reply:
Good to know, thanks for sharing!

Reply:
Sure thing.

> , and I don't see this as a blocker to this PR

I just want to add to this - this *could* be a blocker, but we could perhaps address it by limiting where this connector is available. I'll lean on the @elastic/response-ops-execution team to assess this.